### PR TITLE
This sets up the scaffolding for the `cosigned` CRD types.

### PR DESCRIPTION
**Pull request by mattmoor**
This mostly contains the boilerplate stuff like codegen and whatnot to bootstrap a Knative controller.  This doesn't contain any of the meaningful type definition aspects, which will follow in a subsequent PR.

Signed-off-by: Matt Moore <mattmoor@chainguard.dev>

#### Ticket Link
Sets up for #1417

#### Release Note

```release-note

```

### CODE REVIEW

**Comment by mattmoor** [comment]
cc @kkavitha @coyote240 @hectorj2f 

I'll get this passing CI, but this should set up all of the scaffolding that you need.

**Comment by mattmoor** [comment]
Ultimately, this is the spec for what this CRD will look like in subsequent changes: https://docs.google.com/document/d/1gBLEOOHWOmvHVsoJbgGU74GdwA6CGxMRp3MAeEB50l4/edit

**Comment by dlorenc** [comment]
It verified!!!

**Comment by mattmoor** [comment]
@dlorenc yeah, because `nodiff` needed a `path:` attribute.  Gonna keep poking at it.

File: pkg/apis/cosigned/v1alpha1/clusterimagepolicy_types.go
Comment (mattmoor) [inline pkg/apis/cosigned/v1alpha1/clusterimagepolicy_types.go:54]: This, a body for `SetDefaults` and `Validate` and unit tests are the key bits remaining to close this issue out.

I believe your WIP schema should slot in pretty directly here.

Once this lands, I'm happy to set up the webhook and CRD definition stuff while you sketch the type out, but was aiming for an MVP here to unblock y'all.

cc @kkavitha @coyote240 @hectorj2f

**Review by hectorj2f — APPROVED** [review]
lgtm

File: pkg/apis/cosigned/v1alpha1/clusterimagepolicy_types.go
Comment (kkavitha) [reply pkg/apis/cosigned/v1alpha1/clusterimagepolicy_types.go:54]: awesome! thanks.

**Pull request merged**
